pandas/tests/io/formats/test_format.py

```diff
@@ -3249,14 +3249,24 @@ def test_filepath_or_buffer_arg(
     encoding,
     filepath_or_buffer_id,
 ):
-    if encoding is not None:
-        assert filepath_or_buffer_id is not None
-        assert filepath_or_buffer_id in ["string", "pathlike"]
     df = float_frame
     expected = getattr(df, method)()
 
+    if filepath_or_buffer_id not in ["string", "pathlike"] and encoding is not None:
+        with pytest.raises(
+            ValueError,
+            match="filepath is not a string or path but encoding is specified.",
+        ):
+            raise ValueError(
+                "filepath is not a string or path but encoding is specified."
+            )
```

Comment:
this should be `getattr(df, method)(buf=filepath_or_buffer, encoding=encoding)`

Reply:
Hi, as far as I know, none of `to_html`, `to_latex`, or `to_string` would raise any `ValueError` if this condition is met. So `getattr(df, method)(buf=filepath_or_buffer, encoding=encoding)`
would result in `Failed: DID NOT RAISE <class 'ValueError'>`. How do we handle this?

Reply:
That needs to be added in this PR, xref https://github.com/pandas-dev/pandas/issues/28766#issuecomment-538413490. should be easy once test is written :)

Reply:
Should I do that for all to_string, to_latex and to_html in their respective methods?

Reply:
I _think_ you should be able to add the check in `get_buffer` in `pandas\io\formats\format.py`

Reply:
Cool 😃